shared/src/question-generators/Stack/utilsSequenceLetter.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  FreeTextFeedbackFunction,
+  FreeTextFormatFunction,
+  FreeTextQuestion,
+} from "@shared/api/QuestionGenerator"
+import { insertSpaceAfterEveryXChars } from "@shared/question-generators/huffman-coding/utils/utils"
+import { Stack } from "@shared/question-generators/Stack/Stack"
+import { stackQuestion } from "@shared/question-generators/Stack/StackGenerator"
+import Random from "@shared/utils/random"
+import { t, Translations } from "@shared/utils/translations"
+
+/**
+ * Generates a new Stack question for the seqLetter variant
+ *
+ * Generates a sequence of letters and stars (vja* ad*a s...) (letter means push and * means pop)
+ * Asks the user to enter the sequence of letters that are popped from the stack
+ *
+ * @param lang
+ * @param random
+ * @param permalink
+ * @param translations
+ */
+export function generateVariantSequenceLetter(
+  lang: "de" | "en",
+  random: Random,
+  permalink: string,
+  translations: Translations,
+) {
+  const { sequence, popSequence } = generateOperationsVariantSequenceLetter(random)
+
+  const checkFormat: FreeTextFormatFunction = ({ text }) => {
+    // check if the user input only consists of letters
+    text = text.replace(/\s/g, "")
+    if (!/^[A-Z]*$/.test(text.toUpperCase())) {
+      return {
+        valid: false,
+        message: t(translations, lang, "checkFormatSeqLetter"),
+      }
+    }
+    return {
+      valid: true,
+    }
+  }
+
+  const feedback: FreeTextFeedbackFunction = ({ text }) => {
+    // remove whitespaces and comparing user input (as uppercase) with the correct answer
+    text = text.replace(/\s/g, "")
+    if (text.toUpperCase() === popSequence) {
+      return {
+        correct: true,
+      }
+    }
+    return {
+      correct: false,
+      correctAnswer: popSequence,
+    }
+  }
+
+  const question: FreeTextQuestion = {
+    type: "FreeTextQuestion",
+    name: stackQuestion.name(lang),
+    path: permalink,
+    text: t(translations, lang, "sequenceLetterText", [insertSpaceAfterEveryXChars(sequence, 3)]),
+    checkFormat,
+    feedback,
+  }
+  return { question }
+}
+
+/**
+ * Generates a sequence of letters and stars
+ *
+ * @param random
+ */
+function generateOperationsVariantSequenceLetter(random: Random) {
+  const s = new Stack<string>()
+  const possibleChars: string = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
```

Comment:
Just a thought: It seems that nothing ensures that the solution of the question assembles a word. Maybe we want to restrict the character set. Something like
```
const possibleChars: string = random.choose("ABCDEF", "GHIJKL", "KLMNOP", "UVWXYZ")
```